src/ttf.zig

```diff
@@ -0,0 +1,2818 @@
+///
+/// ## Remarks
+/// This reports the number of times the library has been initialized by a call to `init()`, without a paired deinitialization request from `quit()`.
+///
+/// In short: if it's greater than zero, the library is currently initialized and ready to work. If zero, it is not initialized.
+///
+/// Despite the return value being a signed integer, this function should not return a negative number.
+///
+/// ## Return Value
+/// Returns the current number of initialization calls, that need to eventually be paired with this many calls to `quit()`.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_ttf 3.0.0.
+pub fn wasInit() c_int {
+    return c.TTF_WasInit();
+}
+
+/// Convert from a 4 character string to a 32-bit tag.
+///
+/// ## Function Parameters
+/// * `string`: The 4 character string to convert.
+///
+/// ## Return Value
+/// Returns the 32-bit representation of the string.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Remarks
+/// The string must be 4 characters long.
+///
+/// ## Version
+/// This function is available since SDL_ttf 3.0.0.
+pub fn stringToTag(string: [:0]const u8) u32 {
+    std.debug.assert(string.len == 4); // the string must be 4 characters long.
+    return c.TTF_StringToTag(string.ptr);
+}
+
+/// Convert from a 32-bit tag to a 4 character string.
+///
+/// ## Function Parameters
+/// * `tag`: The 32-bit tag to convert.
+///
+/// ## Return Value
+/// Returns the 4 character representation of the tag.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_ttf 3.0.0.
+pub fn tagToString(tag: u32) [4]u8 {
+    var buf: [5]u8 = undefined;
+    c.TTF_TagToString(tag, &buf, buf.len);
+    return [4]u8{ buf[0], buf[1], buf[2], buf[3] };
+}
+
+/// Get the script used by a 32-bit codepoint.
+///
+/// ## Function Parameters
+/// * `ch`: The character code to check.
+///
+/// ## Return Value
+/// Returns an ISO 15924 code on success.
+///
+/// ## Thread Safety
+/// This function is thread-safe.
+///
+/// ## Version
+/// This function is available since SDL_ttf 3.0.0.
+pub fn getGlyphScript(ch: u32) !u32 {
+    const script = c.TTF_GetGlyphScript(ch);
+    if (script == 0) {
+        errors.callErrorCallback();
+        return error.SdlError;
+    }
+    return script;
+}
+
+pub const font_create_filename_string = c.TTF_PROP_FONT_CREATE_FILENAME_STRING;
+pub const font_create_iostream_pointer = c.TTF_PROP_FONT_CREATE_IOSTREAM_POINTER;
+pub const font_create_iostream_offset_number = c.TTF_PROP_FONT_CREATE_IOSTREAM_OFFSET_NUMBER;
+pub const font_create_iostream_autoclose_boolean = c.TTF_PROP_FONT_CREATE_IOSTREAM_AUTOCLOSE_BOOLEAN;
+pub const font_create_size_float = c.TTF_PROP_FONT_CREATE_SIZE_FLOAT;
+pub const font_create_face_number = c.TTF_PROP_FONT_CREATE_FACE_NUMBER;
```

Comment:
I do not handle properties like this, see a subsystem like `video` to see how properties are done.